5a/Client/src/Program.fs

```diff
@@ -82,17 +85,20 @@ let main _args =
     client.Connect
         (fun error ->
             task {
-                if not (error = null) then
-                    printfn $"Error connecting to cluster: ${error}"
-                    return false
-                else
-                    printfn "Connected to cluster"
-                    return true
+                Async.Sleep(TimeSpan.FromSeconds(1)) |> Async.RunSynchronously
+                attempts <- attempts + 1
```

Comment:
Nah, no need for mutation here :) Let's try to use recursion instead.

Reply:
As discussed, the connect function needs a boolean to be returned in order to know if it should retry or not, which in this case we need to keep state and mutate the `attempts` variable.

Reply:
You were 100% right, I stand corrected!